parsec/backend/invite.py

```diff
@@ -249,6 +249,8 @@ def _to_http_redirection_url(client_ctx, invitation):
         if msg["type"] == InvitationType.USER:
             if client_ctx.profile != UserProfile.ADMIN:
                 return invite_new_serializer.rep_dump({"status": "not_allowed"})
+            if await self.user_already_member(client_ctx.organization_id, msg["claimer_email"]):
```

Comment:
I think we shouldn't have this separated from `new_for_user` given we cannot guarantee atomicity this way

(hence `new_for_user` should raise a new kind of exception to signify the user already exists)

On top that this would remove the need for the `user_already_member` & `retrieve_human` methods (and there memory/postgresql implementations and the abstract declaration) so it should globally simplify this PR ;-)